skimage/measure/fit.py

```diff
@@ -492,66 +487,78 @@ def estimate(self, data):
         success : bool
             True, if model estimation succeeds.
 
-        """
 
+        References:
+        -----------
+        .. [1]  Halir, R.; Flusser, J. "Numerically stable direct least squares
+                fitting of ellipses". In Proc. 6th International Conference in
+                Central Europe on Computer Graphics and Visualization.
+                WSCG (Vol. 98, pp. 125-132).
+
+        """
+        # another REFERENCE: [2] http://mathworld.wolfram.com/Ellipse.html
         _check_data_dim(data, dim=2)
 
         x = data[:, 0]
         y = data[:, 1]
 
-        N = data.shape[0]
-
-        # pre-allocate jacobian for all iterations
-        A = np.zeros((N + 5, 2 * N), dtype=np.double)
-        # same for all iterations: xc, yc
-        A[0, :N] = -1
-        A[1, N:] = -1
-
-        diag_idxs = np.diag_indices(N)
-
-        def fun(params):
-            xyt = self.predict_xy(params[5:], params[:5])
-            fx = x - xyt[:, 0]
-            fy = y - xyt[:, 1]
-            return np.append(fx, fy)
-
-        def Dfun(params):
-            xc, yc, a, b, theta = params[:5]
-            t = params[5:]
-
-            ct = np.cos(t)
-            st = np.sin(t)
-            ctheta = math.cos(theta)
-            stheta = math.sin(theta)
-
-            # derivatives for fx, fy in the following order:
-            #       xc, yc, a, b, theta, t_i
-
-            # fx
-            A[2, :N] = - ctheta * ct
-            A[3, :N] = stheta * st
-            A[4, :N] = a * stheta * ct + b * ctheta * st
-            A[5:, :N][diag_idxs] = a * ctheta * st + b * stheta * ct
-            # fy
-            A[2, N:] = - stheta * ct
-            A[3, N:] = - ctheta * st
-            A[4, N:] = - a * ctheta * ct + b * stheta * st
-            A[5:, N:][diag_idxs] = a * stheta * st - b * ctheta * ct
-
-            return A
-
-        # initial guess of parameters using a circle model
-        params0 = np.empty((N + 5, ), dtype=np.double)
-        xc0 = x.mean()
-        yc0 = y.mean()
-        r0 = np.sqrt((x - xc0)**2 + (y - yc0)**2).mean()
-        params0[:5] = (xc0, yc0, r0, 0, 0)
-        params0[5:] = np.arctan2(y - yc0, x - xc0)
-
-        params, _ = optimize.leastsq(fun, params0, Dfun=Dfun, col_deriv=True)
-
-        self.params = params[:5]
-
+        # Quadratic part of design matrix [eqn. 15] from [1]
+        D1 = np.vstack([x ** 2, x * y, y ** 2]).T
```

Comment:
not really, it this usage the `np.vstack` gives 3 x N and the `np.vstack` gives 1 x 3*N